text/0000-explode.md

```diff
@@ -0,0 +1,236 @@
+- Start Date: 2017-12-07
+- RFC PR: 
+- Ember Issue: 
+
+# Summary
+
+This RFC proposes a way to begin immediately splitting Ember into a family of separate NPM packages without breaking ecosystem compatibility.
+
+# Motivation
+
+This proposal
+
+ - enables people working on small, focused apps to start with a minimal, small core of Ember and only add more features as they are needed
+ 
+ - allows us to more aggressively isolate legacy features by making them pay-as-you-go
+ 
+ - creates an upgrade incentive for apps and addons that is "more carrot, less stick".
+ 
+# Detailed design
+
+## @ember/core Package
+
+As of Ember 3.0, the only supported way to depend on Ember is to depend on the `ember-source` npm package. This proposal does not change the meaning of `ember-source`. It will continue to be all-of-Ember in the way people already understand it, and features can only be removed from `ember-source` at major releases after an appropriate deprecation cycle.
+
+We propose adding an *additional* supported way to depend on Ember. You can choose to run `ember explode`, and we will remove `ember-source` from your dependencies and replace it with `@ember/core` *plus the current complete set of already-extracted Ember packages*. `ember explode` is intended to not change the set of Ember features in your app. It just splats them out into a form that comes from separate packages.
+
+Critically, `@ember/core` includes an extra rule in its semver policy: the only supported way to upgrade to a newer `@ember/core` is to run the `ember-cli-update` command. The update command will be aware of any additional packages have been split off from `@ember/core` since your last version, and it will add them automatically to your app.
+
+This special upgrade requirement gives us the freedom to split off more packages *in any release*, without breaking anybody's app.
```

Comment:
I believe this to be wishful thinking. You'll be able to tell people that it's their own fault to have shot themselves in the foot. But people do expect packages in the npm-ecosystem to adhere to "npm-SemVer", that is, to behave just like any other npm package. Having this special requirement locks us into our own walled SemVer garden. 

Nothing against frequent breaking changes that can be mitigated by a special tool. But we shouldn't redefine what to expect from version numbers in a `package.json`.

Reply:
Maybe a more appropriate, semver-consistent approach would be similar to what this RFC proposes for `ember-source`.

If we want to break up a package, we can leave around the original package as a "meta-package", but it is now possible to opt out of part of it.

@ef4 do you remember if we considered and rejected this when we discussed this?

Reply:
`ember-source` would still follow semver, but would be made up of multiple packages that have their own versions. If a sub-package has a major bump, and if it is able to be integrated into `ember-source` without bumping the major version of `ember-source`, then it will otherwise it would have to wait for a major release to make it in.